### PR TITLE
build: add lint staged

### PR DESCRIPTION
**Pull request by mlaurencin**
Add lint-staged 

cc @codebytere

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -9,8 +9,8 @@
   "scripts": {
     "build": "tsc",
     "start": "probot run ./lib/index.js",
-    "prettier:write": "prettier --write '**/*.ts'",
```
Comment (codebytere) [inline package.json:12]: windows fails with single quotes

**Review by codebytere — APPROVED** [review]

**Review by ckerr — COMMENTED** [review]
Is there an alternate quoting that would work on Windows too, e.g. using double quotes? 

Removing the quotes here works because there aren't many .ts files to be linted here, but on non-Windows we want to ensure that the glob literal is passed to prettier if possible rather than being expanded on the shell. The issue is that if we have a lot of files and expand the glob in the shell, you can hit an E2BIG error in the shell.

For example in build-tools, `prettier --check '**/*.ts'` yields:

```sh
Checking formatting...
All matched files use Prettier code style!
Done in 1.04s.
```

But `yarn prettier --write **/*.ts` yields:

```sh
error An unexpected error occurred: "spawn E2BIG".
info If you think this is a bug, please open a bug report with the information provided in "/home/charles/electron/build-tools/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

_If_ there's an alternative e.g. double quotes that works cross-platform, it might be better to do that so we'll have a one-liner that Just Works in all our repos / on all our platforms. Otherwise, :+1: on dropping the quotes as this PR does, and just having each repo be more selective e.g. making sure it's not linting `node_modules/` by accident :smile:

**Comment by mlaurencin** [comment]
> Is there an alternate quoting that would work on Windows too, e.g. using double quotes?
> 
> Removing the quotes here works because there aren't many .ts files to be linted here, but on non-Windows we want to ensure that the glob literal is passed to prettier if possible rather than being expanded on the shell. The issue is that if we have a lot of files and expand the glob in the shell, you can hit an E2BIG error in the shell.
> 
> For example in build-tools, `prettier --check '**/*.ts'` yields:
> 
> ```shell
> Checking formatting...
> All matched files use Prettier code style!
> Done in 1.04s.
> ```
> 
> But `yarn prettier --write **/*.ts` yields:
> 
> ```shell
> error An unexpected error occurred: "spawn E2BIG".
> info If you think this is a bug, please open a bug report with the information provided in "/home/charles/electron/build-tools/yarn-error.log".
> info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
> ```
> 
> _If_ there's an alternative e.g. double quotes that works cross-platform, it might be better to do that so we'll have a one-liner that Just Works in all our repos / on all our platforms. Otherwise, 👍 on dropping the quotes as this PR does, and just having each repo be more selective e.g. making sure it's not linting `node_modules/` by accident 😄

Double quotes does not seem to work either, so probably the latter has to be done.

**Review by codebytere — APPROVED** [review]
checked locally, also works on macOS!

**Review by ckerr — APPROVED** [review]

**Pull request merged**